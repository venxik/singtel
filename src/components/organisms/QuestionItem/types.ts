import { TouchableOpacityProps } from '../../atoms/Button';

export type QuestionItemProps = {
  text: string;
  isSelected: boolean;
} & TouchableOpacityProps;
