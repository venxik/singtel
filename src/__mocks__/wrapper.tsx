import { Provider } from 'react-redux';
import { AppStore, RootState, setupStore } from '../redux';
import { PreloadedState } from '@reduxjs/toolkit';
import { PropsWithChildren } from 'react';
import { render } from '@testing-library/react-native';

interface ExtendedRenderOptions {
  preloadedState?: PreloadedState<RootState>;
  store?: AppStore;
}

export function AllTheProviders({ children }: { children: JSX.Element }) {
  return <Provider store={setupStore()}>{children}</Provider>;
}

export function renderWithProviders(
  ui: React.ReactElement,
  {
    preloadedState = {},
    store = setupStore(preloadedState),
    ...renderOptions
  }: ExtendedRenderOptions = {}
) {
  function Wrapper({ children }: PropsWithChildren<{}>): JSX.Element {
    return <Provider store={store}>{children}</Provider>;
  }
  return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
}
