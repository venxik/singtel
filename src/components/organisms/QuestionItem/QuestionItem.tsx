import { StyleSheet, TouchableOpacity } from 'react-native';
import { ComicText, useThemeColor } from '../../atoms';
import { QuestionItemProps } from './types';
import React from 'react';

const QuestionItem = (props: QuestionItemProps) => {
  const { style, lightColor, darkColor, isSelected, disabled, text, ...otherProps } = props;

  const backgroundColorSelected = useThemeColor(
    { light: lightColor, dark: darkColor },
    'categorySelected'
  );

  const backgroundColorDisabled = useThemeColor(
    { light: lightColor, dark: darkColor },
    'buttonDisabledBackground'
  );

  return (
    <TouchableOpacity
      style={[
        {
          backgroundColor: disabled ? backgroundColorDisabled : backgroundColorSelected,
          borderRadius: 6,
          justifyContent: 'center',
          width: 50,
          height: 50,
          margin: 4,
        },
        style,
      ]}
      disabled={disabled}
      {...otherProps}>
      <ComicText testID="question-item-text" style={styles.text}>
        {isSelected ? '' : text}
      </ComicText>
    </TouchableOpacity>
  );
};

export default React.memo(QuestionItem);

const styles = StyleSheet.create({
  text: { textAlign: 'center' },
});
