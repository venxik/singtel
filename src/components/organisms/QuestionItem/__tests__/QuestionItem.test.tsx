import React from 'react';
import { renderWithProviders } from '../../../../__mocks__';
import { fireEvent } from '@testing-library/react-native';
import QuestionItem from '../QuestionItem';

describe('QuestionItem', () => {
  it(`match snapshot`, () => {
    const tree = renderWithProviders(<QuestionItem text="test" isSelected />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it(`can be pressed and functional`, () => {
    let number = 1;
    const { getByTestId } = renderWithProviders(
      <QuestionItem
        testID="question-item"
        text="test"
        onPress={() => (number += 1)}
        isSelected={false}
      />
    );
    fireEvent.press(getByTestId('question-item'));
    expect(number).toBe(2);
  });

  it(`can have display text when isSelected false`, () => {
    const { getByTestId } = renderWithProviders(<QuestionItem text="test" isSelected={false} />);
    expect(getByTestId('question-item-text').props.children).toEqual('test');
  });

  it(`doesn't display text when isSelected true`, () => {
    const { getByTestId } = renderWithProviders(<QuestionItem text="test" isSelected={true} />);
    expect(getByTestId('question-item-text').props.children).toEqual('');
  });
});
