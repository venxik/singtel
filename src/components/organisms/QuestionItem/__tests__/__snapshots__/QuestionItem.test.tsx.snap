// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuestionItem match snapshot 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "backgroundColor": "#686fcc",
      "borderRadius": 6,
      "height": 50,
      "justifyContent": "center",
      "margin": 4,
      "opacity": 1,
      "width": 50,
    }
  }
>
  <Text
    style={
      [
        {
          "color": "#000",
        },
        [
          {
            "textAlign": "center",
          },
          {
            "fontFamily": "comic",
          },
        ],
      ]
    }
    testID="question-item-text"
  />
</View>
`;

exports[`StyledText match snapshot 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "backgroundColor": "#686fcc",
      "borderRadius": 6,
      "height": 50,
      "justifyContent": "center",
      "margin": 4,
      "opacity": 1,
      "width": 50,
    }
  }
>
  <Text
    style={
      [
        {
          "color": "#000",
        },
        [
          {
            "textAlign": "center",
          },
          {
            "fontFamily": "comic",
          },
        ],
      ]
    }
    testID="question-item-text"
  />
</View>
`;
